include/MappingResults.hpp

```diff
@@ -21,6 +21,8 @@ struct MappingResults {
     std::size_t   singleQubitGates = 0;
     std::size_t   cnots            = 0;
     std::size_t   layers           = 0;
+    double        totalFidelity    = 1.;
+    double        totalLogFidelity = 0.;
```

Comment:
If I recall correctly, we already discussed this in the last fidelity-awareness PR. I think we settled on this solution, but maybe it is not yet optimal.

The whole reason for tracking both these values but only exposing one of them in the JSON result is about usability and precision. Log fidelity has higher precision (as it uses the whole positive half of the representation space of double, instead of just 0 to 1), while pure fidelity is of course more comprehensible to the average user. For its precision I would prefer to (at least internally) track log fidelity and for the sake of testing I would like to make accessible in C++. But I agreed with you in the last PR, that it would make more sense to only expose pure fidelity to Python and JSON. So this was the result of that discussion.

Reply:
Now that you mention it, I vaguely remember that discussion.
In the end, it's not really too important I guess. We could as well expose both to python. We are talking about a single double 🤷🏽‍♂️
Especially given the argument that one provides more precision than the other, both could be interesting for people.

Reply:
That's true, it's just 1 double. I'll expose both then.